snakemake/cli.py

```diff
@@ -1849,41 +1844,16 @@ def parse_quietness(quietness) -> Set[Quietness]:
 
 
 def setup_log_handlers(args, parser):
```

Comment:
Since we anyway produce a "hidden" breaking change in the API because the objects passed to snakemake.settings.types.OutputSettings.log_handlers are now coming from a different base class, we should probably also change the type there overall and expect a list of strings with logger plugin names in that settings class, while moving the collection of the corresponding plugins into the API classes, analogously to storage and exectuor plugins.

Reply:
Not sure I follow here. Is this what you're thinking:
```diff
@dataclass
class OutputSettings(SettingsBase):
    printshellcmds: bool = False
    nocolor: bool = False
    quiet: Optional[AnySet[Quietness]] = None
    debug_dag: bool = False
    verbose: bool = False
    show_failed_logs: bool = False
-   log_handlers: Sequence[object] = tuple()
+   log_handlers: Sequence[str] = tuple()
    keep_logger: bool = False
    stdout: bool = False
    benchmark_extended: bool = False
```

Reply:
Yes, and the strings are just the names. Then, the instantiation can happen inside of the API classes and the users of the API don't not need to instantiate those things themselves.